examples/other_datasets/plot_financial_data.py

```diff
@@ -0,0 +1,238 @@
+        self.process = process
+        self.n_components = n_components
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X):
+        if not self.process:
+            return X
+        return Whitening(dim_red={"n_components": 4}).fit_transform(X)
+
+
+# Classical pipeline: put together the transformers, and add at the end
+# the classical SVM
+pipe = make_pipeline(
+    ToEpochs(n=10),
+    XdawnCovariances(nfilter=1),
+    OptionalWhitening(process=True, n_components=4),
+    SlimVector(keep_diagonal=True),
+    SVC(),
+)
+
+# Optimize the pipeline:
+# let's save some time and run the optimization with the classical SVM
+gs = GridSearchCV(
+    pipe,
+    param_grid={
+        "toepochs__n": [10, 20],
+        "xdawncovariances__nfilter": [1, 2],
+        "optionalwhitening__process": [True, False],
+        "optionalwhitening__n_components": [2, 4],
+        "slimvector__keep_diagonal": [True, False],
+    },
+    scoring="balanced_accuracy",
+)
+
+
+##############################################################################
+# Balance dataset
+# ---------------
+#
+# Balance the data and run the evaluation on a quantum vs classical pipeline.
+
+# Let's balance the problem using NearMiss.
+# Note: at this stage `features` also contains the `index` column.
+# So `NearMiss` we choose the closest 200 non-fraud epochs to the 200 fraud-epochs
+# based also on this `index` column. This should be improved for real use cases.
+X, y = NearMiss().fit_resample(features.to_numpy(), target.to_numpy())
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+
+labels, counts = np.unique(y_train, return_counts=True)
+print(f"Training set shape: {X_train.shape}, genuine: {counts[0]}, frauds: {counts[1]}")
+
+labels, counts = np.unique(y_test, return_counts=True)
+print(f"Testing set shape: {X_test.shape}, genuine: {counts[0]}, frauds: {counts[1]}")
+
+# before fitting the GridSearchCV, let's display a sample of the epochs:
+epochs = ToEpochs(n=10).transform(X_train)
+print("Profile of an epoch:")
+print(epochs[0])
+
+# ...and the "ERP"
+# (see https://pyriemann.readthedocs.io/en/latest/auto_examples/ERP/plot_ERP.html)
+plot_waveforms(epochs, "hist")
+plt.show()
+
+
+##############################################################################
+# Run evaluation
+# --------------
+
+# Let's fit our GridSearchCV, to find the best hyper parameters
+gs.fit(X_train, y_train)
+
+# Print cross-validation results
+print(gs.cv_results_)
+
+# This is the best score with the classical SVM.
+# (with this train/test split at least)
+score_svm = gs.best_estimator_.score(X_test, y_test)
+
+# Quantum pipeline:
+# let's take the same parameters but evaluate the pipeline with a quantum SVM:
+gs.best_estimator_.steps[4] = ("quanticsvm", QuanticSVM(quantum=True))
+score_qsvm = gs.best_estimator_.fit(X_train, y_train).score(X_test, y_test)
+
+# Print the results
+print(f"Classical: {score_svm} \nQuantum  : {score_qsvm}")
```

Comment:
Quantum pipeline gives a binary classification score of 0,5.
Flipping a coin would do the same thing... unless I missed something.

Reply:
No, this is weird.
In the first version, it was 100%... but with the same data.
I need to investigate this.